### PR TITLE
Upgrade byteorder to 0.5.1

### PR DESCRIPTION
**Pull request by jaysonsantos**
If possible, would you release a new version?

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -10,6 +10,6 @@ readme = "Readme.md"
 keywords = ["image", "metadata"]
 
 [dependencies]
-byteorder = "0.4"
+byteorder = "0.5.*"
```
Comment (netvl) [inline Cargo.toml]: Why the asterisk is necessary?

File: src/common/tiff.rs [hunk 2]
```diff
@@ -404,14 +404,14 @@ pub trait EntryTypeRepr {
     /// Attempts to read the represented value from the given stream with the given byte order.
     ///
     /// Returns the number of bytes read and the value itself.
-    fn read_from<R: Read>(source: &mut R, byte_order: ByteOrder) -> byteorder::Result<(u32, Self::Repr)>;
+    fn read_from<R: Read>(source: &mut R, byte_order: ByteOrder) -> IoResult<(u32, Self::Repr)>;
```
Comment (netvl) [inline src/common/tiff.rs]: I would appreciate if you could change it to `io::Result` instead.

File: Cargo.toml (hunk 1 above)
Comment (jaysonsantos) [inline Cargo.toml]: As they would semantic versioning it would always get the latest bug patches.

File: src/common/tiff.rs (hunk 2 above)
Comment (jaysonsantos) [inline src/common/tiff.rs]: Done

File: Cargo.toml (hunk 1 above)
Comment (netvl) [inline Cargo.toml]: Actually `0.5` would be `^0.5` or, identically, `^0.5.0`, which means any version of `0.5.x` kind. So just `0.5` includes all the patch versions.

Comment (jaysonsantos) [inline Cargo.toml]: Changed back to `0.5` then.

**Comment by netvl** [comment]
Thanks!

**Comment by netvl** [comment]
Available in 0.3.2.

**Comment by jaysonsantos** [comment]
Thank you very much

**Pull request merged**